### PR TITLE
restrict adding new jobs that use legacy clusters

### PR DESCRIPTION
**Pull request by BenTheElder**
as discussed in today's SIG Testing meeting, ahead of the august deadline

... And to ensure this is true, ensure these job tests have restricted approvals as well (they already should have, since they enforce putting jobs for certain clusters in certain directories, but the approvals for these tests were looser than those directories)

/cc @dims @upodroid @aojea @ameukam

### CODE REVIEW

File: config/tests/jobs/OWNERS [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# these tests ensure that jobs are configured safely
+# these tests need to be as tightly approved as config/jobs/image-pushing
+no_parent_owners: true
+approvers:
+# SIG K8s Infra / Testing leads with context
+- dims
+- bentheelder
+- aojea
+- upodroid
```
Comment (BenTheElder) [inline config/tests/jobs/OWNERS:9]: let me alphabetize, add @ameukam and copy to reviewers

File: config/tests/jobs/jobs_test.go [hunk 2]
```diff
@@ -287,6 +287,408 @@ func TestRetestMatchJobsName(t *testing.T) {
...
+func TestCommunityJobs(t *testing.T) {
+	const legacyDefault = "default"
+	const legacyTrusted = "test-infra-trusted"
+
+	// TODO: cut this list down to none
+	// this is a snapshot of docs/job-migration-todo.dm
+	// we are not overcomplicating this as all of them will be removed in august anyhow
+	// tail -n +10 docs/job-migration-todo.md | cut -d \| -f 3 | awk '{ print "\""$0"\","}'
+	knownBadJobs := sets.NewString(
+		"pull-csi-driver-nfs-e2e",
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go:301]: locally tested that this is WAI by commenting some of these out temporarily

Comment (dims) [reply config/tests/jobs/jobs_test.go:301]: ✅

**Pull request merged**